### ISSUE TITLE
Mint 21 Feature Request -- Wayland

### ISSUE THREAD

**Issue by agentl074** [linuxmint/cinnamon#10787]
Apologies if this was already covered -- or is already included in the Ubuntu 22 package base -- however, I would love to see Wayland be the default display server in Mint 21 Cinnamon. 

Thank you all for your hard work on this project!

**Comment by Secret-chest** [comment]
> or is already included in the Ubuntu 22 package base

**Comment by Secret-chest** [comment]
> Thank you...

Sorry... I'm not admin (though I might contribute when I have more time)

**Comment by JosephMcc** [comment]
> Apologies if this was already covered

A search of closed issues will turn up this has been asked and answered plenty. And just to be clear, Wayland will not be used in Mint 21.

**Comment by arjpar** [comment]
Is there an issue for tracking Wayland support progress for Cinnamon?

**Comment by collinss** [comment]
> Is there an issue for tracking Wayland support progress for Cinnamon?

Not really, no.

We still don't have any specific plans for implementing Wayland support. Theoretically, we are a lot closer to achieving it now that muffin has been rebased on a newer version of mutter, but I don't know how much work would still need to be done to make it work, and we're not in a hurry to do so. It's not that we're just dragging our heels either - there are some major downsides still to using wayland. One in particular that we discussed just a couple of weeks ago is that wayland doesn't run in a separate process the way x11 does, so if cinnamon crashes (which can be caused by any number of different things, including a buggy applet), or potentially even if you just restart cinnamon, the entire session is gone, along with all of your open apps (and any unsaved progress). This is a huge regression for us, and while there may be a way to work around this, I suspect that it would require huge architectural changes throughout the project in order to do it, and even then, it may not even be feasible.

Then there's problems like Nvidia support still being subpar (from what I hear), and utilities like redshift and screen recording apps wont work out of the box.

TLDR Wayland isn't ready, and I personally wouldn't expect significant motion on this in the near future.

**Comment by danimesq** [comment]
@collinss
> now that muffin has been rebased on a newer version of mutter,

Is it possible to implement decorations such as blur effect?

**Comment by JosephMcc** [comment]
> Is it possible to implement decorations such as blur effect?

Possible maybe. But the toolkits it's built on simply don't handle blur well and it comes with serious performance issues. Rebasing on a newer Mutter doesn't change that.

**Comment by orowith2os** [comment]
@collinss 

> Not really, no.

I'll have to ask you to reconsider.

> One in particular that we discussed just a couple of weeks ago is that wayland doesn't run in a separate process the way x11 does

This is not true for all compositors; that's one of the beautiful parts of the Wayland protocol. Mutter and wlroots might do this, but kwin doesn't; the shell and compositor are two separate processes, communicating through a custom Wayland protocol. 

You could just as well make your own protocol, similar to KDE's, and use that instead. Bam, you've got an X-like display server design, without much of the additional overhead Xorg has.

> Then there's problems like Nvidia support still being subpar (from what I hear), and utilities like redshift and screen recording apps wont work out of the box.

That is not an issue with Wayland, but with NVIDIA and the utilities themselves. Additionally, there's the Xwayland Video Bridge for compatibility with legacy applications such as Discord that refuse to migrate. There are also already Wayland-compatible alternatives to common Xorg utilities, within reason; see https://arewewaylandyet.com/ for more details. 

NVIDIA lacking support shouldn't be a blocker for implementing Wayland support, or pushing for it on Linux. If anything, we should push for it more, so that NVIDIA has more reason to properly support it.

**Comment by danimesq** [comment]
I don't like conspiracy theories, but be sure @orowith2os:

"challenge Ubuntu but do nothing that could bother Windows" is Linux Mint's main philosophy.
With the advent of DAOs, I seek for it being forked soon.
The Linux Mint project & community has to be saved from the Linux Mint administration.
